### ISSUE TITLE
Markdown Previewer Tests: marked is not defined

### ISSUE THREAD

**Issue by reoams** [freeCodeCamp/testable-projects-fcc#382]
#### Issue Description
When running the unit tests for the markdown previewer, tests 3, 4, 5, 6, ~~and 8~~ all fail with the following error: (*8 shouldn't have been included, as I haven't worked on that yet*)
```
ReferenceError: marked is not defined
    at triggerChange (https://cdn.freecodecamp.org/testable-projects-fcc/v1/bundle.js:20040:20)
    at n.<anonymous> (https://cdn.freecodecamp.org/testable-projects-fcc/v1/bundle.js:20091:10)
    at r (https://cdnjs.cloudflare.com/ajax/libs/mocha/3.0.2/mocha.min.js:2:7852)
    at r.run (https://cdnjs.cloudflare.com/ajax/libs/mocha/3.0.2/mocha.min.js:2:8853)
    at i.runTest (https://cdnjs.cloudflare.com/ajax/libs/mocha/3.0.2/mocha.min.js:2:13553)
    at https://cdnjs.cloudflare.com/ajax/libs/mocha/3.0.2/mocha.min.js:2:14192
    at r (https://cdnjs.cloudflare.com/ajax/libs/mocha/3.0.2/mocha.min.js:2:13024)
    at https://cdnjs.cloudflare.com/ajax/libs/mocha/3.0.2/mocha.min.js:2:13000
    at n (https://cdnjs.cloudflare.com/ajax/libs/mocha/3.0.2/mocha.min.js:2:12791)
    at https://cdnjs.cloudflare.com/ajax/libs/mocha/3.0.2/mocha.min.js:2:12855
    at i (https://cdnjs.cloudflare.com/ajax/libs/mocha/3.0.2/mocha.min.js:1:559)
```
I am using the CDN test suite bundle `https://cdn.freecodecamp.org/testable-projects-fcc/v1/bundle.js` and bootstrapped my project with the create-react-app.  I have added the marked library as a dependency, and have imported it into my Preview component.  I believe this is an issue when using the CDN test suite on a non-CodePen project, as the tests pass when I put my code in a CodePen project.  The app works identically on both CodePen and locally.

#### Browser Information

* Browser Name, Version:  Chrome v67.0.3396.87 (Official Build) (64-bit); Firefox v60.0.2 (64-bit)
* Operating System: Window 7 Professional, Service Pack 1
* Mobile, Desktop, or Tablet: Desktop

#### Your Code / Link to Your Pen
[CodePen Project](https://codepen.io/elrolfe/pen/eKMaXa)

Code from [create-react-app bootstrapped project](https://gitlab.com/elrolfe-fcc-frontend/markdown)

App.jsx
```
import React, { Component } from 'react';
import Preview from "./Preview";

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      markdown: "# Header\n## Sub Header\n[Link](https://www.freecodecamp.org)\nInline `code` markup\n```\nCode Block\n```\n\nUnordered List\n- Item\n- Item\n- Item\n  - Sub Item\n- Item\n\nOrdered List\n1. Item\n2. Item\n3. Item\n  1. Ordered Sub Item\n4. Item\n\n> Block quote\n\n**Bold Text**\n\n![alt text](https://s19.postimg.cc/4c5sakssz/profile.jpg)"
    };

    this.updateMarkdown = this.updateMarkdown.bind(this);
  }

  updateMarkdown(event) {
    this.setState({
      markdown: event.target.value
    });
  }

  render() {
    return (
      <div className="App">
        <h1>Markdown Previewer</h1>
        <div className="rows">
          <div className="wrapper">
            <p className="header">Editor</p>
            <textarea id="editor" value={this.state.markdown} onChange={this.updateMarkdown}></textarea>
          </div>
          <div className="wrapper">
            <p className="header">Preview</p>
            <Preview markdown={this.state.markdown} />
          </div>
        </div>
      </div>
    );
  }
}

export default App;
```

Preview.jsx
```
import React, { Component } from "react";

const marked = require("marked");

class Preview extends Component {
  markdown() {
    return {
      __html: marked(this.props.markdown)
    };
  }

  render() {
    return (
      <div id="preview" dangerouslySetInnerHTML={ this.markdown() }></div>
    );
  }
}

export default Preview;
```

**Comment by reoams** [comment]
@tbushman Thanks for the suggestion.  I tried including marked in the App component, but was still running into errors.  I was able to come up with a workaround though.  Adding the marked CDN script to the index.html file put the marked library in the window object.  After doing that, I was able to take out the require in my Preview component, and access the marked library through window.marked instead.  I haven't taken the time to play around much more with it, but I'm guessing that if I had assigned the marked library to the window object after requiring it in my component, it probably would have had the same effect.

I think it's interesting that the tests are depending on the marked library being installed on the window object by the app being tested.  Yes, it's strongly suggested that you use the library, but only in hints.  It's never specifically required by the User Stories.  If someone decided to ignore the hint and roll their own markdown processor (which I may very well have done in my younger and stupider days of wanting to do everything myself), they would never pass the tests.
